### PR TITLE
Bridge rebased to latest from main

### PR DESCRIPTION
**Pull request by dmetzgar**
Includes Stephen's duplex tests, copyright info, and appdomain fixes.

### CODE REVIEW

**Comment by dmetzgar** [comment]
Rebased to main again and squashed all the changes, including the removal of Newtonsoft.Json from all the outerloop tests and the dependency on VS2015 for building the bridge.

File: tools/Bridge/Bridge/App.config [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <startup> 
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.1" />
+    </startup>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
```
Comment (SajayAntony) [inline tools/Bridge/Bridge/App.config]: Can remove I think.

File: tools/Bridge/Bridge/Bridge.csproj [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      <HintPath>$(PackageOutputDir)\Microsoft.Owin.Host.HttpListener.3.0.1\lib\net45\Microsoft.Owin.Host.HttpListener.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Owin.Hosting, Version=3.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>$(PackageOutputDir)\Microsoft.Owin.Hosting.3.0.1\lib\net45\Microsoft.Owin.Hosting.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>$(PackageOutputDir)\Newtonsoft.Json.6.0.6\lib\net45\Newtonsoft.Json.dll</HintPath>
+      <Private>True</Private>
```
Comment (SajayAntony) [inline tools/Bridge/Bridge/Bridge.csproj]: Do you still need this?

**Comment by SajayAntony** [comment]
Once you get the buddy build done by Ron I think we should merge it :shipit:

**Pull request merged**